benchmarks/project/src/main/java/io/zeebe/Worker.java

```diff
@@ -56,7 +57,7 @@ public void run() {
                   requestFutures.add(
                       jobClient
                           .newCompleteCommand(job.getKey())
-                          .variables(job.getVariables())
+                          .variables(variablesToCompleteJobWith)
```

Comment:
Hm this would change our default logic right?

Reply:
Yes, but only when there is a race condition, otherwise this doesn't change anything. Secondly, it is bad practice to complete jobs with all received variables because of this anyways.

Reply:
how is this a race condition? :sweat_smile: You set it to empty now per default :thinking:

Reply:
There isn't now, but if this task is parallelised and there is some other change made to those variables in the mean time then it is. That is as well the only change in logic that is possible. Currently, all received variables are sent along with the complete job command and merged into (i.e. overwritten) in the workflow instance. So nothing really happens, because they are overwritten with the same value. We consider this bad practice because it can easily lead to race conditions when you parallelise part of the workflow. 

Completing the job with no variables, just means that no variables will be merged into the workflow instance

Reply:
Please lets discuss via zoom